driver-core/src/test/java/com/datastax/driver/core/ProtocolBetaVersionTest.java

```diff
@@ -15,48 +15,109 @@
  */
 package com.datastax.driver.core;
 
-import org.junit.experimental.runners.Enclosed;
-import org.junit.runner.RunWith;
+import com.datastax.driver.core.exceptions.NoHostAvailableException;
+import com.datastax.driver.core.exceptions.UnsupportedProtocolVersionException;
 import org.testng.annotations.Test;
 
-import static org.junit.Assert.assertEquals;
+import static com.datastax.driver.core.ProtocolVersion.V4;
+import static com.datastax.driver.core.ProtocolVersion.V5;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.fail;
 
-@RunWith(Enclosed.class)
-public class ProtocolBetaVersionTest {
+/**
+ * Tests for the new USE_BETA flag introduced in protocol v5
+ * and Cassandra 3.10.
+ */
+@CCMConfig(createCluster = false, version = "3.10")
+public class ProtocolBetaVersionTest extends CCMTestsSupport {
 
-    public static class TestAutoUpgrade extends BetaVersionTest {
-        @Override
-        public Cluster.Builder createClusterBuilder() {
-            return super.createClusterBuilder().setAllowBetaProtocolVersions(false);
-        }
+    /**
+     * Verifies that the driver can connect to 3.10 with the following combination of options:
+     * Version V5
+     * Flag SET
+     * Expected version: V5
+     *
+     * @jira_ticket JAVA-1248
+     */
+    @Test(groups = "short")
+    public void should_connect_with_beta_when_beta_version_explicitly_required_and_flag_set() throws Exception {
```

Comment:
Isn't there now no difference between this test and `should_connect_with_beta_when_no_version_explicitly_required_and_flag_set()`?  I think technically this test can be deleted now that the protocol version should not be provided with `allowBetaProtocolVersion`